File: api/schemas/roster.py
from pydantic import BaseModel


class FloorUserOut(BaseModel):
    id: int
    full_name: str
    username: str
    role: str
    shift: str


class ProvisionRequest(BaseModel):
    full_name: str
    email: str
    username: str
    pin: str
    role: str  # "operator" | "packer" - self-registration-style pages never grant above this
    shift: str
    target_lph: float = 400.0


class ResetPinRequest(BaseModel):
    user_id: int
    new_pin: str

File: frontend/src/api/roster.ts
import { api } from './client'

export interface FloorUser {
  id: number
  full_name: string
  username: string
  role: string
  shift: string
}

export const rosterApi = {
  list: () => api.get<FloorUser[]>('/roster'),
  provision: (body: {
    full_name: string; email: string; username: string; pin: string
    role: 'operator' | 'packer'; shift: string; target_lph?: number
  }) => api.post<{ ok: boolean }>('/roster/provision', body),
  resetPin: (userId: number, newPin: string) =>
    api.post<{ ok: boolean }>('/roster/reset-pin', { user_id: userId, new_pin: newPin }),
}
